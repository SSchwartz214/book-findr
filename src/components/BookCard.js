import React from "react";

const BookCard = props => {
  let cleanAuthors;

  if (props.authors) {
    cleanAuthors = props.authors.map((author, i, array) => {
      return i === array.length - 1 ? (
        <span key={`authors-${i}`}>{author}</span>
      ) : (
        <span key={`authors-${i}`}>{author}, </span>
      );
    });
  } else {
    cleanAuthors = "Author not available";
  }

  return (
    <div className="card-container">
      <img src={props.image} alt="" />
      <div className="desc">
        <h2>{props.title}</h2>
        <h3>{cleanAuthors}</h3>
        <p>Publisher: {props.publisher}</p>
        <a href={props.link} target="_blank" rel="noopener noreferrer">
          Read More
        </a>
      </div>
    </div>
  );
};

export default BookCard;
