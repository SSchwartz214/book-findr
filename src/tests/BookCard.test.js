import React from "react";
import { shallow } from "enzyme";
import BookCard from "../components/BookCard";

describe("BookCard", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <BookCard
        key={1}
        image={"wwww.image.com"}
        title={"Ruby"}
        authors={["Seth"]}
        publisher={"SethCo"}
        link={"wwww.link.com"}
      />
    );
  });

  it("should match snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should match snapshot", () => {
    wrapper = shallow(
      <BookCard
        key={1}
        image={"wwww.image.com"}
        title={"Ruby"}
        authors={["Seth", "Joe"]}
        publisher={"SethCo"}
        link={"wwww.link.com"}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it("should match snapshot", () => {
    wrapper = shallow(
      <BookCard
        key={1}
        image={"wwww.image.com"}
        title={"Ruby"}
        authors={null}
        publisher={"SethCo"}
        link={"wwww.link.com"}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
