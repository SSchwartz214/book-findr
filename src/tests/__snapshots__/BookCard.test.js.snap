// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookCard should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BookCard
    authors={
      Array [
        "Seth",
      ]
    }
    image="wwww.image.com"
    link="wwww.link.com"
    publisher="SethCo"
    title="Ruby"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          alt=""
          src="wwww.image.com"
        />,
        <div
          className="desc"
        >
          <h2>
            Ruby
          </h2>
          <h3>
            <span>
              Seth
            </span>
          </h3>
          <div>
            Publisher: 
            <div
              dangerouslySetInnerHTML={
                Object {
                  "__html": "SethCo",
                }
              }
            />
          </div>
          <a
            href="wwww.link.com"
            rel="noopener noreferrer"
            target="_blank"
          >
            Read more
          </a>
        </div>,
      ],
      "className": "card-container",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "",
          "src": "wwww.image.com",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h2>
              Ruby
            </h2>,
            <h3>
              <span>
                Seth
              </span>
            </h3>,
            <div>
              Publisher: 
              <div
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "SethCo",
                  }
                }
              />
            </div>,
            <a
              href="wwww.link.com"
              rel="noopener noreferrer"
              target="_blank"
            >
              Read more
            </a>,
          ],
          "className": "desc",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Ruby",
            },
            "ref": null,
            "rendered": "Ruby",
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <span>
                  Seth
                </span>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "authors-0",
                "nodeType": "host",
                "props": Object {
                  "children": "Seth",
                },
                "ref": null,
                "rendered": "Seth",
                "type": "span",
              },
            ],
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Publisher: ",
                <div
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "SethCo",
                    }
                  }
                />,
              ],
            },
            "ref": null,
            "rendered": Array [
              "Publisher: ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "dangerouslySetInnerHTML": Object {
                    "__html": "SethCo",
                  },
                },
                "ref": null,
                "rendered": null,
                "type": "div",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Read more",
              "href": "wwww.link.com",
              "rel": "noopener noreferrer",
              "target": "_blank",
            },
            "ref": null,
            "rendered": "Read more",
            "type": "a",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt=""
            src="wwww.image.com"
          />,
          <div
            className="desc"
          >
            <h2>
              Ruby
            </h2>
            <h3>
              <span>
                Seth
              </span>
            </h3>
            <div>
              Publisher: 
              <div
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "SethCo",
                  }
                }
              />
            </div>
            <a
              href="wwww.link.com"
              rel="noopener noreferrer"
              target="_blank"
            >
              Read more
            </a>
          </div>,
        ],
        "className": "card-container",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "",
            "src": "wwww.image.com",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h2>
                Ruby
              </h2>,
              <h3>
                <span>
                  Seth
                </span>
              </h3>,
              <div>
                Publisher: 
                <div
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "SethCo",
                    }
                  }
                />
              </div>,
              <a
                href="wwww.link.com"
                rel="noopener noreferrer"
                target="_blank"
              >
                Read more
              </a>,
            ],
            "className": "desc",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Ruby",
              },
              "ref": null,
              "rendered": "Ruby",
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <span>
                    Seth
                  </span>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": "authors-0",
                  "nodeType": "host",
                  "props": Object {
                    "children": "Seth",
                  },
                  "ref": null,
                  "rendered": "Seth",
                  "type": "span",
                },
              ],
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Publisher: ",
                  <div
                    dangerouslySetInnerHTML={
                      Object {
                        "__html": "SethCo",
                      }
                    }
                  />,
                ],
              },
              "ref": null,
              "rendered": Array [
                "Publisher: ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "dangerouslySetInnerHTML": Object {
                      "__html": "SethCo",
                    },
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "div",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Read more",
                "href": "wwww.link.com",
                "rel": "noopener noreferrer",
                "target": "_blank",
              },
              "ref": null,
              "rendered": "Read more",
              "type": "a",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`BookCard should match snapshot 2`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BookCard
    authors={
      Array [
        "Seth",
        "Joe",
      ]
    }
    image="wwww.image.com"
    link="wwww.link.com"
    publisher="SethCo"
    title="Ruby"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          alt=""
          src="wwww.image.com"
        />,
        <div
          className="desc"
        >
          <h2>
            Ruby
          </h2>
          <h3>
            <span>
              Seth
              , 
            </span>
            <span>
              Joe
            </span>
          </h3>
          <div>
            Publisher: 
            <div
              dangerouslySetInnerHTML={
                Object {
                  "__html": "SethCo",
                }
              }
            />
          </div>
          <a
            href="wwww.link.com"
            rel="noopener noreferrer"
            target="_blank"
          >
            Read more
          </a>
        </div>,
      ],
      "className": "card-container",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "",
          "src": "wwww.image.com",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h2>
              Ruby
            </h2>,
            <h3>
              <span>
                Seth
                , 
              </span>
              <span>
                Joe
              </span>
            </h3>,
            <div>
              Publisher: 
              <div
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "SethCo",
                  }
                }
              />
            </div>,
            <a
              href="wwww.link.com"
              rel="noopener noreferrer"
              target="_blank"
            >
              Read more
            </a>,
          ],
          "className": "desc",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Ruby",
            },
            "ref": null,
            "rendered": "Ruby",
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <span>
                  Seth
                  , 
                </span>,
                <span>
                  Joe
                </span>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "authors-0",
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Seth",
                    ", ",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "Seth",
                  ", ",
                ],
                "type": "span",
              },
              Object {
                "instance": null,
                "key": "authors-1",
                "nodeType": "host",
                "props": Object {
                  "children": "Joe",
                },
                "ref": null,
                "rendered": "Joe",
                "type": "span",
              },
            ],
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Publisher: ",
                <div
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "SethCo",
                    }
                  }
                />,
              ],
            },
            "ref": null,
            "rendered": Array [
              "Publisher: ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "dangerouslySetInnerHTML": Object {
                    "__html": "SethCo",
                  },
                },
                "ref": null,
                "rendered": null,
                "type": "div",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Read more",
              "href": "wwww.link.com",
              "rel": "noopener noreferrer",
              "target": "_blank",
            },
            "ref": null,
            "rendered": "Read more",
            "type": "a",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt=""
            src="wwww.image.com"
          />,
          <div
            className="desc"
          >
            <h2>
              Ruby
            </h2>
            <h3>
              <span>
                Seth
                , 
              </span>
              <span>
                Joe
              </span>
            </h3>
            <div>
              Publisher: 
              <div
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "SethCo",
                  }
                }
              />
            </div>
            <a
              href="wwww.link.com"
              rel="noopener noreferrer"
              target="_blank"
            >
              Read more
            </a>
          </div>,
        ],
        "className": "card-container",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "",
            "src": "wwww.image.com",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h2>
                Ruby
              </h2>,
              <h3>
                <span>
                  Seth
                  , 
                </span>
                <span>
                  Joe
                </span>
              </h3>,
              <div>
                Publisher: 
                <div
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "SethCo",
                    }
                  }
                />
              </div>,
              <a
                href="wwww.link.com"
                rel="noopener noreferrer"
                target="_blank"
              >
                Read more
              </a>,
            ],
            "className": "desc",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Ruby",
              },
              "ref": null,
              "rendered": "Ruby",
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <span>
                    Seth
                    , 
                  </span>,
                  <span>
                    Joe
                  </span>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": "authors-0",
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "Seth",
                      ", ",
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    "Seth",
                    ", ",
                  ],
                  "type": "span",
                },
                Object {
                  "instance": null,
                  "key": "authors-1",
                  "nodeType": "host",
                  "props": Object {
                    "children": "Joe",
                  },
                  "ref": null,
                  "rendered": "Joe",
                  "type": "span",
                },
              ],
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Publisher: ",
                  <div
                    dangerouslySetInnerHTML={
                      Object {
                        "__html": "SethCo",
                      }
                    }
                  />,
                ],
              },
              "ref": null,
              "rendered": Array [
                "Publisher: ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "dangerouslySetInnerHTML": Object {
                      "__html": "SethCo",
                    },
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "div",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Read more",
                "href": "wwww.link.com",
                "rel": "noopener noreferrer",
                "target": "_blank",
              },
              "ref": null,
              "rendered": "Read more",
              "type": "a",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`BookCard should match snapshot 3`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BookCard
    authors={null}
    image="wwww.image.com"
    link="wwww.link.com"
    publisher="SethCo"
    title="Ruby"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          alt=""
          src="wwww.image.com"
        />,
        <div
          className="desc"
        >
          <h2>
            Ruby
          </h2>
          <h3>
            Author not available
          </h3>
          <div>
            Publisher: 
            <div
              dangerouslySetInnerHTML={
                Object {
                  "__html": "SethCo",
                }
              }
            />
          </div>
          <a
            href="wwww.link.com"
            rel="noopener noreferrer"
            target="_blank"
          >
            Read more
          </a>
        </div>,
      ],
      "className": "card-container",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "",
          "src": "wwww.image.com",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h2>
              Ruby
            </h2>,
            <h3>
              Author not available
            </h3>,
            <div>
              Publisher: 
              <div
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "SethCo",
                  }
                }
              />
            </div>,
            <a
              href="wwww.link.com"
              rel="noopener noreferrer"
              target="_blank"
            >
              Read more
            </a>,
          ],
          "className": "desc",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Ruby",
            },
            "ref": null,
            "rendered": "Ruby",
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Author not available",
            },
            "ref": null,
            "rendered": "Author not available",
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Publisher: ",
                <div
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "SethCo",
                    }
                  }
                />,
              ],
            },
            "ref": null,
            "rendered": Array [
              "Publisher: ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "dangerouslySetInnerHTML": Object {
                    "__html": "SethCo",
                  },
                },
                "ref": null,
                "rendered": null,
                "type": "div",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Read more",
              "href": "wwww.link.com",
              "rel": "noopener noreferrer",
              "target": "_blank",
            },
            "ref": null,
            "rendered": "Read more",
            "type": "a",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt=""
            src="wwww.image.com"
          />,
          <div
            className="desc"
          >
            <h2>
              Ruby
            </h2>
            <h3>
              Author not available
            </h3>
            <div>
              Publisher: 
              <div
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "SethCo",
                  }
                }
              />
            </div>
            <a
              href="wwww.link.com"
              rel="noopener noreferrer"
              target="_blank"
            >
              Read more
            </a>
          </div>,
        ],
        "className": "card-container",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "",
            "src": "wwww.image.com",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h2>
                Ruby
              </h2>,
              <h3>
                Author not available
              </h3>,
              <div>
                Publisher: 
                <div
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "SethCo",
                    }
                  }
                />
              </div>,
              <a
                href="wwww.link.com"
                rel="noopener noreferrer"
                target="_blank"
              >
                Read more
              </a>,
            ],
            "className": "desc",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Ruby",
              },
              "ref": null,
              "rendered": "Ruby",
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Author not available",
              },
              "ref": null,
              "rendered": "Author not available",
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Publisher: ",
                  <div
                    dangerouslySetInnerHTML={
                      Object {
                        "__html": "SethCo",
                      }
                    }
                  />,
                ],
              },
              "ref": null,
              "rendered": Array [
                "Publisher: ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "dangerouslySetInnerHTML": Object {
                      "__html": "SethCo",
                    },
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "div",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Read more",
                "href": "wwww.link.com",
                "rel": "noopener noreferrer",
                "target": "_blank",
              },
              "ref": null,
              "rendered": "Read more",
              "type": "a",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
